core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala

```diff
@@ -61,11 +68,11 @@ private[ui] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app")
             <li><strong>User:</strong> {app.desc.user}</li>
             <li><strong>Cores:</strong>
             {
-              if (app.desc.maxCores.isEmpty) {
+              if (maxCores.isEmpty) {
                 "Unlimited (%s granted)".format(app.coresGranted)
               } else {
                 "%s (%s granted, %s left)".format(
-                  app.desc.maxCores.get, app.coresGranted, app.coresLeft)
+                  maxCores.get, app.coresGranted, maxCores.get - app.coresGranted)
```

Comment:
note: `maxCores.get - app.coresGranted` may be negative before we kill the executors.